EIPS/eip-6188.md

```diff
@@ -0,0 +1,48 @@
+---
+eip: 6188
+title: Nonce Cap
+description: Caps the nonce at 2^64-2
+author: Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/eip-6190-functional-selfdestruct/12232
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-12-20
+requires: 2929
+---
+
+## Abstract
+
+This EIP caps the nonce at `2^64-2`, reserving it for contracts with unusual behavior, as defined in other EIPs.
+
+## Motivation
+
+This EIP is not terribly useful on its own, as it adds additional computation without any useful side effects. However, it can be used by other EIPs.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### EOA Transactions
+
+The nonce of a transaction originating from an EOA MUST be less than `2^64-2`. If the nonce is either `2^64-1` or `2^64-2`, the transaction MUST be invalid.
+
+### `CREATE` and `CREATE2`
+
+If a nonce would be incremented to `2^64-1` by `CREATE` or `CREATE2`, it is instead set to `2^64-2`. `2^64-1` is reserved for alias or other special contracts.
+
+## Rationale
+
+Capping a nonce allows for contracts with special properties to be created, with their functionality based on their contract code. This EIP is not terribly useful on its own, but it can be used by other EIPs.
+
+## Backwards Compatibility
+
+This EIP requires a protocol upgrade, since it modifies consensus rules. The further restriction of nonce should not have an effect on accounts, as reaching a nonce of `2^64-2` is unfeasible.
+
+## Security Considerations
+
+None.
```

Comment:
If a contract uses `CREATE`, and it somehow was at the nonce cap, and the target address had self-destructed, would further CREATEs put the deployed contract in an unexpected location (i.e. back at the same address instead of at a different address)? Should CREATE fail in that scenario instead? Probably worth mentioning, even if the solution is just "no contract will get to that nonce."